public/pages/createDetector/hooks/useFetchDetectorInfo.ts

```diff
@@ -55,5 +60,6 @@ export const useFetchDetectorInfo = (
   return {
     detector: detector || {},
     hasError: !isEmpty(hasError) && isEmpty(detector),
+    isLoadingDetector: isDetectorRequesting || isIndicesRequesting,
```

Comment:
Not blocker: previously we have seen one [issue](https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/pull/112/files) where `isDetectorRequesting` is true while no other request is going, but we haven't figured out what is the root cause. I think for now we may need to keep eye on the DetectorDetail page to see if there is any weird thing after change is deployed to test domain.

Reply:
I see, ok. When making these changes I've noticed issues with error handling as well. I'm pretty sure it is related to the unhandled Promise rejection thrown [here](https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/blob/master/server/routes/ad.ts#L468) when there is an error retrieving a detector from the backend (invalid id, timeout, etc.). I'm thinking that this could be causing the request to never finish, and leave the ad state's `requesting` field to true. 

I will need to write up an issue to document this as well

Reply:
Example: when changing the url for detector configuration page to an invalid detector id: page is permanently in loading state and chrome devtools shows this: 
![Screen Shot 2020-05-18 at 6 07 25 PM](https://user-images.githubusercontent.com/62119629/82273537-937bcc80-9932-11ea-998e-9adf7b91604a.png)

Reply:
Cool, let's create separate issue to track.

Reply:
Detailed explanation here: https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/issues/157

Reply:
cool. thanks for doing that.